program/shinyApp/server.R

```diff
@@ -577,22 +636,207 @@ server <- function(input,output,session){
+            }
+          )
+          output$DownloadUpdatedSampleTable <- downloadHandler(
+            filename = function() {
+              paste0("updated_SampleTable", Sys.Date(), ".csv") # Dynamic filename
+            },
+            content = function(file) {
+              write.csv(sample_table, file, row.names = TRUE) # Save the reactive data with row names
+            }
+          )
+          output$DownloadUpdatedEntitieAnnotation <- downloadHandler(
+            filename = function() {
+              paste0("updated_EntitieAnno", Sys.Date(), ".csv") # Dynamic filename
+            },
+            content = function(file) {
+              write.csv(annotation_rows, file, row.names = TRUE) # Save the reactive data with row names
+            }
+          )
+          observeEvent(input$close_modal_VI, {
+            removeModal()  # Close the modal
+            # Redo relevant checks
+            check1 <- ifelse(all(rownames(Matrix) == rownames(annotation_rows)),snippetYes,snippetNo)
+            check2 <- ifelse(all(colnames(Matrix) == rownames(sample_table)),snippetYes,snippetNo)
+            check6 <- ifelse(all(colnames(Matrix) == colnames(Matrix)),snippetYes,snippetNo)
+            # Add any additional actions you want to perform
+            shinyjs::click(id = "inspect_data")
+            output$DataMatrix_VI <- DT::renderDataTable({DT::datatable(data = Matrix)})
+            output$SampleMatrix_VI <- DT::renderDataTable({DT::datatable(data = sample_table)})
+            output$EntitieMatrix_VI <- DT::renderDataTable({
+              DT::datatable(data = annotation_rows)
+            })
+            showNotification("Check if now you pass all tests")
+            output$OverallChecks <- renderText({
+              paste0(
+                "Some overall Checks have been run:\n",
+                "Data Matrix is a real csv (has ',' as separators:): ",check0,"\n",
+                "Data Matrix has only numeric values: ",check7,"\n",
+                "Rownames of Matrix are the same as rownames of entitie table ",check1,"\n",
+                "Colnames of Matrix are same as rownames of sample table ",check2," \n",
+                "Matrix has no na (missing values) ",check3,"\n",
+                "Sample table no na (missing values) ",check4,"\n",
+                "Entitie table no na  (missing values) ",check5,"\n",
+                "Sample IDs have valid names ", check6, "\n"
+              )
+            })
+            if(grepl(snippetYes,check0) & 
+               grepl(snippetYes,check1) & 
+               grepl(snippetYes,check2) & 
+               grepl(snippetYes,check3) & 
+               grepl(snippetYes,check4) & # not crucial
+              # grepl(snippetYes,check5) & # not crucial
+               grepl(snippetYes,check6) &
+               grepl(snippetYes,check7)){
+              res_tmp[[session$token]]$passedVI <<- T
+            } else {
+              res_tmp[[session$token]]$passedVI <<- F
+            }
```

Comment:
I think this check also belongs to line 555. Otherwise the test always fail

Reply:
not sure what you ment, did insert it though